applications/grpc_wallet/src/grpc_server.rs

```diff
@@ -0,0 +1,496 @@
+                let sn = c.screen_name.clone();
+                let address = format!("{}", c.address.clone());
+                if let Ok(pk) = c.pub_key.to_base64() {
+                    contacts_resp.push(ContactRpc {
+                        screen_name: sn,
+                        pub_key: pk,
+                        address,
+                    });
+                }
+            }
+        }
+
+        future::ok(Response::new(ContactsRpc {
+            contacts: contacts_resp,
+        }))
+    }
+}
+
+/// Utility function to sort out the interrim TextMessages Collection into an RPC response
+pub fn sort_text_messages(msgs: TextMessages) -> TextMessagesResponseRpc {
+    let mut body = TextMessagesResponseRpc {
+        sent_messages: Vec::new(),
+        received_messages: Vec::new(),
+    };
+
+    for m in msgs.received_messages {
+        match ReceivedTextMessageRpc::try_from(m) {
+            Ok(m) => body.received_messages.push(m),
+            Err(e) => error!("Error serializing text messages: {:?}", e),
+        }
+    }
+
+    // TODO TextMessageService will be refactored to use the boolean `acknowledged` flag rather than two
+    // lists when the Sqlite backend is put in place
+    for msg in msgs.pending_messages {
+        let source = match msg.source_pub_key.to_base64() {
+            Ok(pk) => pk,
+            Err(e) => {
+                error!(target: LOG_TARGET, "Error encoding Public Key to Base 64: {:?}", e);
+                continue;
+            },
+        };
+        let dest = match msg.dest_pub_key.to_base64() {
+            Ok(pk) => pk,
+            Err(e) => {
+                error!(target: LOG_TARGET, "Error encoding Public Key to Base 64: {:?}", e);
+                continue;
+            },
+        };
+        body.sent_messages.push(SentTextMessageRpc {
+            id: msg.id,
+            source_pub_key: source,
+            dest_pub_key: dest,
+            message: msg.message,
+            timestamp: msg.timestamp.to_string(),
+            acknowledged: false,
+        });
+    }
+
+    for msg in msgs.sent_messages {
+        let source = match msg.source_pub_key.to_base64() {
+            Ok(pk) => pk,
+            Err(e) => {
+                error!(target: LOG_TARGET, "Error encoding Public Key to Base 64: {:?}", e);
+                continue;
+            },
+        };
+        let dest = match msg.dest_pub_key.to_base64() {
+            Ok(pk) => pk,
+            Err(e) => {
+                error!(target: LOG_TARGET, "Error encoding Public Key to Base 64: {:?}", e);
+                continue;
+            },
+        };
+        body.sent_messages.push(SentTextMessageRpc {
+            id: msg.id,
+            source_pub_key: source,
+            dest_pub_key: dest,
+            message: msg.message,
+            timestamp: msg.timestamp.to_string(),
+            acknowledged: true,
+        });
+    }
+    body
+}
+
+pub struct WalletServerConfig {
```

Comment:
Maybe the WalletServerConfig and WalletServer should be in their own file `wallet_server.rs`

Reply:
Ja I agree, I think that will be neater. I might then rename this file grpc_interface as it wont have the server stuff in it.